gpflow/inducing_variables/inducing_variables.py

```diff
@@ -45,7 +49,7 @@ def __init__(self, Z: TensorData, name: Optional[str] = None):
         self.Z = Parameter(Z, dtype=default_float())
```

Comment:
```suggestion
        if not isinstance(Z, tf.Variable):
            Z = tf.Variable(Z, dtype=default_float())
        self.Z = Z
```
would allow the test (and users like @maltamiranomontero in #1578) to use
`InducingPoints(tf.Variable(Z, shape=(None, D)))` instead of having to overwrite the `Z` attribute after constructing the InducingPoints instance. Thoughts?

Reply:
Seems like a good idea frame like that but then, tbh, I don't really understand what the potential drawbacks would be. Are all the potential concrete types in 'TensorData' supported by 'tf.variable'?